archive/notifier/src/main/scala/uk/ac/wellcome/platform/archive/notifier/Main.scala

```diff
@@ -1,43 +1,74 @@
 package uk.ac.wellcome.platform.archive.notifier
 
-import com.google.inject.{Guice, Injector}
+import java.net.URL
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import com.amazonaws.services.sns.AmazonSNS
+import com.amazonaws.services.sqs.AmazonSQSAsyncClient
+import com.google.inject.{AbstractModule, Guice, Injector}
 import grizzled.slf4j.Logging
-import uk.ac.wellcome.platform.archive.notifier.modules.{
-  AppConfigModule,
-  ConfigModule
-}
-import uk.ac.wellcome.platform.archive.common.modules.{
-  AkkaModule,
-  CloudWatchClientModule,
-  SNSClientModule,
-  SQSClientModule
-}
+import uk.ac.wellcome.messaging.sns.SNSConfig
+import uk.ac.wellcome.messaging.sqs.SQSConfig
+import uk.ac.wellcome.monitoring.MetricsSender
+import uk.ac.wellcome.platform.archive.common.modules._
 
-import scala.concurrent.Await
+import scala.concurrent.{Await, Future}
 import scala.concurrent.duration.Duration
 
-object Main extends App with Logging {
-  val injector: Injector = Guice.createInjector(
-    new AppConfigModule(args),
-    ConfigModule,
+object Main extends App {
+  val app = new NotifierApp()
+  app.run()
+}
+
+class NotifierApp() extends WellcomeApp {
+  val configModule = TypesafeConfigModule
```

Comment:
maybe if you use `lazy val` here and in line 28? I think in general we should find solutions where we don't set uninitialised vals in traits to override them in extending classes. It's brittle and often leads to weird initialisation errors